sky/serve/replica_managers.py

```diff
@@ -271,6 +304,15 @@ def to_replica_status(self) -> serve_state.ReplicaStatus:
             # Pending to launch
             return serve_state.ReplicaStatus.PENDING
         if self.sky_launch_status == ProcessStatus.RUNNING:
```

Comment:
Actually, should we set the `self.sky_launch_status` to `INTERRUPTED`?

Reply:
maybe should not modify status given `to_replica_status`

Reply:
We should modify `to_replica_status` as well

Reply:
Oh I meant should we set it under `to_replica_status`, because the name suggests it converts (launch/down) status property to replica status.

Reply:
Oh I mean we should set the `self.sky_launch_status` to `INTERRUPTED` in preemption handling or any other places that could trigger such scale-down when provisioning, and in `to_replica_status` we use `self.sky_launch_status == INTERRUPTED` to judge, instead of here we use `sky_launch_status == RUNNING`

Reply:
oh I see. I misunderstood what you meant. Added